src/main.cpp

```diff
@@ -110,6 +131,42 @@ int main(int argc, char** argv)
             }
             defines.push_back(define);
         }
+        else if (a.rfind("-W", 0) == 0)
+        {
+            std::string option;
+            bool enabled;
+            if (a.rfind("-Wno-", 0) == 0)
+            {
+                enabled = false;
+                option = a.substr(5);
+            }
+            else
+            {
+                enabled = true;
+                option = a.substr(2);
+            }
+            Warning warning = parse_warning_option(option);
+            if (warning == Unknown)
+            {
+                fprintf(
+                    stderr, "error: unknown warning option '%s'\n", a.c_str());
```

Comment:
The usage needs to list the possible warnings.

Reply:
Yes, I'm consider doing that. Also, I'm thinking updating the existing warnings in a separated PR. What do you think?

Reply:
Sounds good.